web/packages/api/src/toEthereum_v2.ts

```diff
@@ -347,38 +385,97 @@ export async function getDeliveryFee(
-        "0x0000000000000000000000000000000000000000",
-        "0x0000000000000000000000000000000000000000",
-        "0x0000000000000000000000000000000000000000000000000000000000000000",
-        340282366920938463463374607431768211455n,
-        340282366920938463463374607431768211455n,
-        340282366920938463463374607431768211455n,
-        parachain,
-        340282366920938463463374607431768211455n
-    )
+    const { tokenErcMetadata, sourceParachain, ahAssetMetadata, sourceAssetMetadata } =
+        resolveInputs(registry, tokenAddress, parachain)
 
-    let assetHubExecutionFeeDOT = 0n
-    let returnToSenderExecutionFeeDOT = 0n
-    let returnToSenderDeliveryFeeDOT = 0n
-    const bridgeHubDeliveryFeeDOT = await calculateDeliveryFee(
-        assetHub,
-        registry.bridgeHubParaId,
-        xcm
-    )
-    if (parachain !== registry.assetHubParaId) {
-        const returnToSenderXcm = buildParachainERC20ReceivedXcmOnDestination(
+    let xcm: any, forwardedXcm: any
+
+    if (sourceAssetMetadata.location) {
+        xcm = buildResultXcmAssetHubPNATransferFromParachain(
+            assetHub.registry,
+            registry.ethChainId,
+            sourceAssetMetadata.locationOnAH,
+            sourceAssetMetadata.locationOnEthereum,
+            "0x0000000000000000000000000000000000000000000000000000000000000000",
+            "0x0000000000000000000000000000000000000000",
+            "0x0000000000000000000000000000000000000000000000000000000000000000",
+            340282366920938463463374607431768211455n,
+            340282366920938463463374607431768211455n,
+            340282366920938463463374607431768211455n
+        )
+        forwardedXcm = buildExportXcmForPNA(
             assetHub.registry,
             registry.ethChainId,
+            sourceAssetMetadata.locationOnEthereum,
+            "0x0000000000000000000000000000000000000000",
+            "0x0000000000000000000000000000000000000000000000000000000000000000",
+            340282366920938463463374607431768211455n,
+            340282366920938463463374607431768211455n,
+            1000
+        )
+    } else {
+        xcm = buildResultXcmAssetHubERC20TransferFromParachain(
+            assetHub.registry,
+            registry.ethChainId,
+            "0x0000000000000000000000000000000000000000000000000000000000000000",
             "0x0000000000000000000000000000000000000000",
+            "0x0000000000000000000000000000000000000000",
+            "0x0000000000000000000000000000000000000000000000000000000000000000",
+            340282366920938463463374607431768211455n,
             340282366920938463463374607431768211455n,
             340282366920938463463374607431768211455n,
+            parachain,
+            340282366920938463463374607431768211455n
+        )
+        forwardedXcm = buildExportXcmForERC20(
+            assetHub.registry,
+            registry.ethChainId,
+            tokenAddress,
```

Comment:
Do we need this? Can we not just pass in empty token for estimation?
```suggestion
            "0x0000000000000000000000000000000000000000",
```

Reply:
Same as https://github.com/Snowfork/snowbridge/pull/1419#discussion_r2026997065?

Reply:
But this introduces a new parameter to getting the fee estimate. Not sure if we really need it.